### PR TITLE
CPLAT-7110 Run the unit tests using dart2js in Dart 2

### PR DESCRIPTION
**Pull request by todbachman-wf**
## Motivation
We typically use w_module in web apps so it would be nice to confirm the code works properly when compiled using dart2js.

## Changes
Updated CI to use dart2js when running the unit tests with Dart 2. I also simplified how we submit code coverage reports to codecov in the Dockerfile.

#### Release Notes
Use dart2js when running unit tests with Dart 2 in CI.

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - Passing CI should be sufficient
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/w_module/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/w_module/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,6 @@ cache:
     - $HOME/.pub-cache
 script:
   - pub get
-  - pub run dart_dev dart2-only -- format --check
+  - pub run dart_dev format --check
```
Comment (smaifullerton-wk) [inline .travis.yml]: Won't this fail if the code is formatted with Dart 1?

Comment (smaifullerton-wk) [reply .travis.yml]: Nvm. It's already formatted with Dart 2

File: tool/dev.dart [hunk 2]
```diff
@@ -32,5 +33,8 @@ Future<Null> main(List<String> args) async {
   config.copyLicense.directories = directories;
   config.format.paths = directories;
 
+  config.test
+    ..platforms = Platform.version.startsWith('2') ? ['chrome'] : ['vm'];
```
Comment (corwinsheahan-wf) [inline tool/dev.dart]: In Dart 1 we run tests on the vm?

Comment (todbachman-wf) [reply tool/dev.dart]: Currently I believe we are. So I thought I'd leave it on the VM. I could change that to content shell if that would be preferable. I don't have a strong feeling about it.

**Comment by corwinsheahan-wf** [comment]
QA +1
* CI Passes is sufficient for test only changes.

Merging into master.
@Workiva/release-management-p

**Pull request merged**